### DISCUSSION TITLE
Enabling --strip-comments=true by default

### DISCUSSION THREAD

**Discussion by unforgettableid** [jgm/pandoc#10344] [Polls]
Hi!  The current default is `--strip-comments=false`.

Like @ousia suggested [elsewhere](https://github.com/jgm/pandoc/issues/2552#issuecomment-174308710):

I think the ideal default would be `--strip-comments=true`.  That way, HTML comments would be automatically removed for new users by default.  They wouldn't have to look online in order to learn about the `--strip-comments` option and how to use it.

In the unlikely case that a user wanted to pass HTML comments through to the output file verbatim:  They could manually specify `--strip-comments=false`.

Thank you for reading this, and for all the work you've done on Pandoc!

-----

Dear all:  Feel free to agree or disagree, by clicking the GitHub "thumbs up" or "thumbs down" emoji buttons.  You can also comment below, if you'd like.

-----

**Edit:**  I know that there are [better ways](https://stackoverflow.com/questions/4823468/comments-in-markdown) to add comments to a Markdown file, instead of using HTML comment tags.  But many people don't know about these better ways.  Therefore, I still think it would be best to make `--strip-comments=true` the default.

**Comment by tarleb** [comment]
I've converted this into a discussion, because the primary point should be to gather feedback.

My personal opinion is that the current behavior should be kept. Pandoc tries to preserve information as well as possible, and that includes all HTML snippets, including comments. Changing that would bend pandoc's principles.
